src/MongoDB.Driver.Core/Core/Operations/RetryabilityHelper.cs

```diff
@@ -86,9 +86,16 @@ static RetryabilityHelper()
         }
 
         // public static methods
-        public static void AddRetryableWriteErrorLabelIfRequired(MongoException exception)
+        public static void AddRetryableWriteErrorLabelIfRequired(MongoException exception, SemanticVersion serverVersion)
         {
-            if (ShouldRetryableWriteExceptionLabelBeAdded(exception))
+            if (!Feature.RetryableWrites.IsSupported(serverVersion))
+            {
+                return;
+            }
+
+            if (IsNetworkException(exception) ||
```

Comment:
according to spec, the driver should add an error label regardless server version (if >=3.6)

Reply:
Put all these new conditions inside `ShouldRetryableWriteExceptionLabelBeAdded`.

Change old line 91 (new lines 91-98) to:

    if (ShouldRetryableWriteExceptionLabelBeAdded(exception, serverVersion))

Reply:
done